src/apis/conversations/conversations.repository.ts

```diff
@@ -1,13 +1,43 @@
-import { BaseAbstractRepostitory } from '@app/common/base/base.repository';
+import { BaseAbstractRepository } from '@app/common/base/base.repository';
 import { Injectable } from '@nestjs/common';
 import { ConversationEntity } from './entity/conversations.entity';
 import { ConversationsRepositoryInterface } from './conversations.interface';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { Repository, SelectQueryBuilder } from 'typeorm';
 
 @Injectable()
-export class ConversationRepository extends BaseAbstractRepostitory<ConversationEntity> implements ConversationsRepositoryInterface {
-  constructor(@InjectRepository(ConversationEntity) private readonly userRepository: Repository<ConversationEntity>) {
-    super(userRepository);
+export class ConversationRepository extends BaseAbstractRepository<ConversationEntity> implements ConversationsRepositoryInterface {
+  constructor(@InjectRepository(ConversationEntity)
+  private readonly ConversationRepository: Repository<ConversationEntity>) {
+    super(ConversationRepository);
   }
+
+  generalQueryBuilder(): SelectQueryBuilder<ConversationEntity> {
+    return this.ConversationRepository.createQueryBuilder('conversations');
+  }
+
+  async findConversationByName(name: string): Promise<ConversationEntity[]> {
+    const query = this.generalQueryBuilder();
+    if (name) {
+      query.where('conversations.name LIKE :name', { name: `%${name}%` });
+    }
+    query.andWhere('conversations.isActive IN (true, false)', { isActive: [true, false] });
```

Comment:
explain for me write code here

Reply:
this code get conversation with isActive True and False(if hasn't  andWhere is get with isActive True )